src/nupic/regions/SDRClassifierRegion.py

```diff
@@ -0,0 +1,502 @@
+    proto.steps = self.steps
+    proto.alpha = self.alpha
+    proto.verbosity = self.verbosity
+    proto.maxCategoryCount = self.maxCategoryCount
+
+    self._sdrClassifier.write(proto.sdrClassifier)
+
+
+  @classmethod
+  def readFromProto(cls, proto):
+    """Read state from proto object.
+
+    proto: CLAClassifierRegionProto capnproto object
+    """
+    instance = cls()
+
+    instance.classifierImp = proto.classifierImp
+    instance.steps = proto.steps
+    instance.alpha = proto.alpha
+    instance.verbosity = proto.verbosity
+    instance.maxCategoryCount = proto.maxCategoryCount
+
+    instance._sdrClassifier = SDRClassifierFactory.read(proto)
+
+    return instance
+
+
+  def reset(self):
+    pass
+
+
+  def compute(self, inputs, outputs):
+    """
+    Process one input sample.
+    This method is called by the runtime engine.
+    @param inputs -- inputs of the classifier region
+    @param outputs -- outputs of the classifier region
+
+    """
+
+    # This flag helps to prevent double-computation, in case the deprecated 
+    # customCompute() method is being called in addition to compute() called 
+    # when network.run() is called
+    self._computeFlag = True
+
+    # An input can potentially belong to multiple categories. 
+    # If a category value is < 0, it means that the input does not belong to 
+    # that category.
+    categories = [category for category in inputs["categoryIn"]
+                  if category >= 0]
+
+    activeCells = inputs["bottomUpIn"]
+    patternNZ = activeCells.nonzero()[0]
+
+    # ==========================================================================
+    # Allow to train on multiple input categories. 
+    # Do inference first, and then train on all input categories.
+
+    # --------------------------------------------------------------------------
+    #   1. Call classifier. Don't train. Just inference. Train after.
+
+    # Dummy classification input, because this param is required. Learning is 
+    # off, so the classifier is not learning this input. Inference only here.
+    classificationIn = {"actValue": 0, "bucketIdx": 0}
+    clResults = self._sdrClassifier.compute(recordNum=self.recordNum,
+                                            patternNZ=patternNZ,
+                                            classification=classificationIn,
+                                            learn=False,
+                                            infer=self.inferenceMode)
+
+    for category in categories:
+      classificationIn = {"bucketIdx": int(category), "actValue": int(category)}
+
+      # ------------------------------------------------------------------------
+      #   2. Train classifier, no inference
+      self._sdrClassifier.compute(recordNum=self.recordNum,
+                                  patternNZ=patternNZ,
+                                  classification=classificationIn,
+                                  learn=self.learningMode,
```

Comment:
What happens if self.learningMode is False? Is it still correct to call compute here?